src/main/java/org/dependencytrack/integrations/defectdojo/DefectDojoClient.java

```diff
@@ -68,4 +72,104 @@ public void uploadDependencyTrackFindings(final String token, final String engag
             uploader.handleUnexpectedHttpResponse(LOGGER, request.getUrl(), response.getStatus(), response.getStatusText());
         }
     }
+
+    // Pulling DefectDojo 'tests' API endpoint, and retrieve a list of existing tests
+    public ArrayList getDojoTestIds(final String token) {
+        LOGGER.debug("Pulling DefectDojo Tests API ...");
+        String tests_uri = "/api/v2/tests/";
+        final UnirestInstance ui = UnirestFactory.getUnirestInstance();
+        LOGGER.debug("Make the first pagination call");
+        HttpResponse<JsonNode> response = ui.get(baseURL + tests_uri + "?limit=100")
```

Comment:
This will retrieve ALL tests in defect dojo, which could be thousands. You can filter on the desired engagment by specifying a `engagement=<id>`. But as mentioned in my other comment, all this logic is not needed IIRC.

Reply:
Good point. I can add the optimization in like you suggest. Thank you.

Reply:
Change with suggested improvement is now in.